model/operations.py

```diff
@@ -175,6 +176,24 @@ def knockout_undo(model, to_undo):
             model.reactions.get_by_id(r_id).knock_out()
     return removed
 
+def changebounds_undo(model, to_undo):
+    for rn in to_undo:
+        if model.reactions.has_id(rn['id']):
+            model.reactions.get_by_id(rn['id']).bounds = \
+                rn['lower_bound'], rn['upper_bound']
+    return {reaction['id'] for reaction in to_undo}
+
+
+async def changebounds_apply(model, to_apply):
+    changed = []
+    before = {r['id'] for r in model.notes['changes']['changed']['reactions']}
+    for rn in to_apply:
+        changed.append(reaction_to_dict(model.reactions.get_by_id(rn['id'])))
+        model.reactions.get_by_id(rn['id']).bounds = rn['bounds'][0], rn['bounds'][1]
```

Comment:
not necessary to extract the items with the index like that, `model.reactions.get_by_id(rn['id']).bounds = rn['bounds']` is enough